--- xbmc-9.11-b1/configure.in.wiget	2009-12-08 20:04:30.000000000 +0100
+++ xbmc-9.11-b1/configure.in	2009-12-08 20:13:04.000000000 +0100
@@ -334,41 +334,41 @@ AC_CHECK_LIB([z],           [main],, AC_
 AC_CHECK_LIB([ssl],         [main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([crypto],      [main],, AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([MAD],        [mad],
-                  [INCLUDES+=" $MAD_CFLAGS"; LIBS+=" $MAD_LIBS"],
+                  [INCLUDES="${INCLUDES} $MAD_CFLAGS"; LIBS="${LIBS} $MAD_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([FONTCONFIG], [fontconfig],
-                  [INCLUDES+=" $FONTCONFIG_CFLAGS"; LIBS+=" $FONTCONFIG_LIBS"],
+                  [INCLUDES="${INCLUDES} $FONTCONFIG_CFLAGS"; LIBS="${LIBS} $FONTCONFIG_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([FRIBIDI],    [fribidi],
-                  [INCLUDES+=" $FRIBIDI_CFLAGS"; LIBS+=" $FRIBIDI_LIBS"],
+                  [INCLUDES="${INCLUDES} $FRIBIDI_CFLAGS"; LIBS="${LIBS} $FRIBIDI_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([SQLITE3],    [sqlite3],
-                  [INCLUDES+=" $SQLITE3_CFLAGS"; LIBS+=" $SQLITE3_LIBS"],
+                  [INCLUDES="${INCLUDES} $SQLITE3_CFLAGS"; LIBS="${LIBS} $SQLITE3_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([PNG],        [libpng],
-                  [INCLUDES+=" $PNG_CFLAGS"; LIBS+=" $PNG_LIBS"],
+                  [INCLUDES="${INCLUDES} $PNG_CFLAGS"; LIBS="${LIBS} $PNG_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([PCRE],       [libpcre],
-                  [INCLUDES+=" $PCRE_CFLAGS"; LIBS+=" $PCRE_LIBS"]; \
+                  [INCLUDES="${INCLUDES} $PCRE_CFLAGS"; LIBS="${LIBS} $PCRE_LIBS"]; \
                   AC_DEFINE([HAVE_LIBPCRE],[1],["Define to 1 if libpcre is installed"]),
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([CDIO],       [libcdio],
-                  [INCLUDES+=" $CDIO_CFLAGS"; LIBS+=" $CDIO_LIBS"],
+                  [INCLUDES="${INCLUDES} $CDIO_CFLAGS"; LIBS="${LIBS} $CDIO_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([SAMPLERATE], [samplerate],
-                  [INCLUDES+=" $SAMPLERATE_CFLAGS"; LIBS+=" $SAMPLERATE_LIBS"],
+                  [INCLUDES="${INCLUDES} $SAMPLERATE_CFLAGS"; LIBS="${LIBS} $SAMPLERATE_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([MMS],        [libmms],
-                  [INCLUDES+=" $MMS_CFLAGS"; LIBS+=" $MMS_LIBS"],
+                  [INCLUDES="${INCLUDES} $MMS_CFLAGS"; LIBS="${LIBS} $MMS_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([FREETYPE2],  [freetype2],
-                  [INCLUDES+=" $FREETYPE2_CFLAGS"; LIBS+=" $FREETYPE2_LIBS"],
+                  [INCLUDES="${INCLUDES} $FREETYPE2_CFLAGS"; LIBS="${LIBS} $FREETYPE2_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([OGG],        [ogg],
-                  [INCLUDES+=" $OGG_CFLAGS"; LIBS+=" $OGG_LIBS"],
+                  [INCLUDES="${INCLUDES} $OGG_CFLAGS"; LIBS="${LIBS} $OGG_LIBS"],
                   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([VORBIS],     [vorbis],
-                  [INCLUDES+=" $VORBIS_CFLAGS"; LIBS+=" $VORBIS_LIBS"],
+                  [INCLUDES="${INCLUDES} $VORBIS_CFLAGS"; LIBS="${LIBS} $VORBIS_LIBS"],
                   AC_MSG_ERROR($missing_library))
 
 # platform dependent libraries
@@ -376,10 +376,10 @@ if test "$host_vendor" = "apple" ; then
   AC_CHECK_LIB([iconv],     [main],, AC_MSG_ERROR($missing_library))
   AC_CHECK_LIB([smbclient], [main],, AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([VORBISFILE],     [vorbisfile],
-                    [INCLUDES+=" $VORBISFILE_CFLAGS"; LIBS+=" $VORBISFILE_LIBS"],
+                    [INCLUDES="${INCLUDES} $VORBISFILE_CFLAGS"; LIBS="${LIBS} $VORBISFILE_LIBS"],
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([VORBISENC],     [vorbisenc],
-                    [INCLUDES+=" $VORBISENC_CFLAGS"; LIBS+=" $VORBISENC_LIBS"],
+                    [INCLUDES="${INCLUDES} $VORBISENC_CFLAGS"; LIBS="${LIBS} $VORBISENC_LIBS"],
                     AC_MSG_ERROR($missing_library))
 else
   AC_CHECK_LIB([faad],      [main],, AC_MSG_ERROR($missing_library))
@@ -394,37 +394,37 @@ else
   AC_CHECK_LIB([SDL_image], [main],, AC_MSG_ERROR($missing_library))
   
   PKG_CHECK_MODULES([ALSA],       [alsa],
-                  [INCLUDES+=" $ALSA_CFLAGS"; LIBS+=" $ALSA_LIBS"],
+                  [INCLUDES="${INCLUDES} $ALSA_CFLAGS"; LIBS="${LIBS} $ALSA_LIBS"],
                   AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([ENCA],         [enca],
-                    [INCLUDES+=" $ENCA_CFLAGS"; LIBS+=" $ENCA_LIBS"],
+                    [INCLUDES="${INCLUDES} $ENCA_CFLAGS"; LIBS="${LIBS} $ENCA_LIBS"],
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([XT],           [xt],
-                    [INCLUDES+=" $XT_CFLAGS"; LIBS+=" $XT_LIBS"],
+                    [INCLUDES="${INCLUDES} $XT_CFLAGS"; LIBS="${LIBS} $XT_LIBS"],
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([XTST],         [xtst],
-                    [INCLUDES+=" $XTST_CFLAGS"; LIBS+=" $XTST_LIBS"],
+                    [INCLUDES="${INCLUDES} $XTST_CFLAGS"; LIBS="${LIBS} $XTST_LIBS"],
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([XMU],          [xmu],
-                    [INCLUDES+=" $XMU_CFLAGS"; LIBS+=" $XMU_LIBS"],
+                    [INCLUDES="${INCLUDES} $XMU_CFLAGS"; LIBS="${LIBS} $XMU_LIBS"],
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([XINERAMA],     [xinerama],
-                    [INCLUDES+=" $XINERAMA_CFLAGS"; LIBS+=" $XINERAMA_LIBS"],
+                    [INCLUDES="${INCLUDES} $XINERAMA_CFLAGS"; LIBS="${LIBS} $XINERAMA_LIBS"],
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([CURL],         [libcurl],
-                    [INCLUDES+=" $CURL_CFLAGS"; LIBS+=" $CURL_LIBS"],
+                    [INCLUDES="${INCLUDES} $CURL_CFLAGS"; LIBS="${LIBS} $CURL_LIBS"],
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([DBUS],         [dbus-1],
-                    [INCLUDES+=" $DBUS_CFLAGS"; LIBS+=" $DBUS_LIBS"],
+                    [INCLUDES="${INCLUDES} $DBUS_CFLAGS"; LIBS="${LIBS} $DBUS_LIBS"],
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([HAL_STORAGE],  [hal-storage],
-                    [INCLUDES+=" $HAL_STORAGE_CFLAGS"; LIBS+=" $HAL_STORAGE_LIBS"], 
+                    [INCLUDES="${INCLUDES} $HAL_STORAGE_CFLAGS"; LIBS="${LIBS} $HAL_STORAGE_LIBS"], 
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([SDL],          [sdl],
-                    [INCLUDES+=" $SDL_CFLAGS"; LIBS+=" $SDL_LIBS"],
+                    [INCLUDES="${INCLUDES} $SDL_CFLAGS"; LIBS="${LIBS} $SDL_LIBS"],
                     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([HAL],          [hal],
-                    [INCLUDES+=" $HAL_CFLAGS"; LIBS+=" $HAL_LIBS"],
+                    [INCLUDES="${INCLUDES} $HAL_CFLAGS"; LIBS="${LIBS} $HAL_LIBS"],
                     AC_MSG_ERROR($missing_library))
 fi
 
@@ -1053,13 +1053,14 @@ AC_DEFUN([XB_CONFIG_MODULE],[
 AC_CONFIG_COMMANDS_POST([
 if [[ $3 != "1" ]]; then
     if [[ -d $1 ]]; then
-        pushd $1
+	OLDPWD=$(pwd)    
+        cd $1
         $2
         if [[ $? -ne 0 ]]; then
-            popd
+            cd $OLDPWD
             AC_MSG_ERROR([[Submodule $1 failed to configure]])
         else
-            popd
+            cd $OLDPWD
         fi
     else
         AC_MSG_ERROR([[Submodule $1 doesn't exist]])
