diff -urN xbmc-10.0.orig/configure.in xbmc-10.0/configure.in
--- xbmc-10.0.orig/configure.in	2010-12-21 18:31:22.222999999 +0100
+++ xbmc-10.0/configure.in	2010-12-21 19:07:25.468000001 +0100
@@ -248,7 +248,7 @@
 
 AC_ARG_ENABLE([hal],
   [AS_HELP_STRING([--disable-hal],
-  [disable HAL support (default is enabled if hal and hal-storage is found)])],
+  [disable HAL support (default is enabled if hal and hal-storage is founC_CONFIG_COMMANDS_POSTd)])],
   [use_hal=$enableval],
   [use_hal=yes])
 
@@ -553,48 +553,48 @@
 AC_CHECK_LIB([mpeg2convert],[main],, AC_MSG_ERROR($missing_library))
 AC_CHECK_LIB([ssh],         [sftp_tell64],, AC_MSG_RESULT([Could not find suitable version of libssh]))
 PKG_CHECK_MODULES([MAD],        [mad],
-  [INCLUDES="$INCLUDES $MAD_CFLAGS"; LIBS="$LIBS $MAD_LIBS"],
+  [INCLUDES="${INCLUDES} $MAD_CFLAGS"; LIBS="${LIBS} $MAD_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([FONTCONFIG], [fontconfig],
-  [INCLUDES="$INCLUDES $FONTCONFIG_CFLAGS"; LIBS="$LIBS $FONTCONFIG_LIBS"],
+  [INCLUDES="$INCLUDES $FONTCONFIG_CFLAGS"; LIBS="${LIBS} $FONTCONFIG_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([FRIBIDI],    [fribidi],
-  [INCLUDES="$INCLUDES $FRIBIDI_CFLAGS"; LIBS="$LIBS $FRIBIDI_LIBS"],
+  [INCLUDES="${INCLUDES} $FRIBIDI_CFLAGS"; LIBS="${LIBS} $FRIBIDI_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([SQLITE3],    [sqlite3],
-  [INCLUDES="$INCLUDES $SQLITE3_CFLAGS"; LIBS="$LIBS $SQLITE3_LIBS"],
+  [INCLUDES="${INCLUDES} $SQLITE3_CFLAGS"; LIBS="${LIBS} $SQLITE3_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([PNG],        [libpng],
-  [INCLUDES="$INCLUDES $PNG_CFLAGS"; LIBS="$LIBS $PNG_LIBS"],
+  [INCLUDES="${INCLUDES} $PNG_CFLAGS"; LIBS="${LIBS} $PNG_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([PCRE],       [libpcre],
-  [INCLUDES="$INCLUDES $PCRE_CFLAGS"; LIBS="$LIBS $PCRE_LIBS"]; \
+  [INCLUDES="${INCLUDES} $PCRE_CFLAGS"; LIBS="${LIBS} $PCRE_LIBS"]; \
   AC_DEFINE([HAVE_LIBPCRE],[1],["Define to 1 if libpcre is installed"]),
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([PCRECPP],    [libpcrecpp],
-  [INCLUDES="$INCLUDES $PCRECPP_CFLAGS"; LIBS="$LIBS $PCRECPP_LIBS"]; \
+  [INCLUDES="${INCLUDES} $PCRECPP_CFLAGS"; LIBS="${LIBS} $PCRECPP_LIBS"]; \
   AC_DEFINE([HAVE_LIBPCRECPP],[1],["Define to 1 if libpcrecpp is installed"]),
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([CDIO],       [libcdio],
-  [INCLUDES="$INCLUDES $CDIO_CFLAGS"; LIBS="$LIBS $CDIO_LIBS"],
+  [INCLUDES="${INCLUDES} $CDIO_CFLAGS"; LIBS="${LIBS} $CDIO_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([SAMPLERATE], [samplerate],
-  [INCLUDES="$INCLUDES $SAMPLERATE_CFLAGS"; LIBS="$LIBS $SAMPLERATE_LIBS"],
+  [INCLUDES="${INCLUDES} $SAMPLERATE_CFLAGS"; LIBS="${LIBS} $SAMPLERATE_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([MMS],        [libmms],
-  [INCLUDES="$INCLUDES $MMS_CFLAGS"; LIBS="$LIBS $MMS_LIBS"],
+  [INCLUDES="${INCLUDES} $MMS_CFLAGS"; LIBS="${LIBS} $MMS_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([FREETYPE2],  [freetype2],
-  [INCLUDES="$INCLUDES $FREETYPE2_CFLAGS"; LIBS="$LIBS $FREETYPE2_LIBS"],
+  [INCLUDES="${INCLUDES} $FREETYPE2_CFLAGS"; LIBS="${LIBS} $FREETYPE2_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([OGG],        [ogg],
-  [INCLUDES="$INCLUDES $OGG_CFLAGS"; LIBS="$LIBS $OGG_LIBS"],
+  [INCLUDES="${INCLUDES} $OGG_CFLAGS"; LIBS="${LIBS} $OGG_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([VORBIS],     [vorbis],
-  [INCLUDES="$INCLUDES $VORBIS_CFLAGS"; LIBS="$LIBS $VORBIS_LIBS"],
+  [INCLUDES="${INCLUDES} $VORBIS_CFLAGS"; LIBS="${LIBS} $VORBIS_LIBS"],
   AC_MSG_ERROR($missing_library))
 PKG_CHECK_MODULES([VORBISENC],     [vorbisenc],
-  [INCLUDES="$INCLUDES $VORBISENC_CFLAGS"; LIBS="$LIBS $VORBISENC_LIBS"],
+  [INCLUDES="${INCLUDES} $VORBISENC_CFLAGS"; LIBS="${LIBS} $VORBISENC_LIBS"],
   AC_MSG_ERROR($missing_library))
 
 # check for libbluray
@@ -608,7 +608,7 @@
 
 AS_CASE([x$use_libbluray],
   [xyes],[
-    INCLUDES="$INCLUDES $LIBBLURAY_CFLAGS";
+    INCLUDES="${INCLUDES} $LIBBLURAY_CFLAGS";
     if test "$host_vendor" != "apple"; then
       XB_FIND_SONAME([BLURAY], [bluray], [use_libbluray])
     fi;
@@ -626,13 +626,13 @@
   AC_CHECK_LIB([mysqlclient], [main],, AC_MSG_ERROR($missing_library))
   AC_CHECK_LIB([smbclient], [main],, AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([VORBISFILE],     [vorbisfile],
-    [INCLUDES="$INCLUDES $VORBISFILE_CFLAGS"; LIBS="$LIBS $VORBISFILE_LIBS"],
+    [INCLUDES="${INCLUDES} $VORBISFILE_CFLAGS"; LIBS="${LIBS} $VORBISFILE_LIBS"],
     AC_MSG_ERROR($missing_library))
 else
   AC_CHECK_PROG(MYSQL_CONFIG, mysql_config, "yes", "no")
   if test $MYSQL_CONFIG = "yes"; then
-    INCLUDES="$INCLUDES `mysql_config --include`"
-    LIBS="$LIBS `mysql_config --libs`"
+    INCLUDES="${INCLUDES} `mysql_config --include`"
+    LIBS="${LIBS} `mysql_config --libs`"
   else
     AC_MSG_ERROR($missing_program)
   fi
@@ -652,34 +652,34 @@
   AC_CHECK_LIB([SDL_image],  [main],, AC_MSG_ERROR($missing_library))
   
   PKG_CHECK_MODULES([ALSA],  [alsa],
-    [INCLUDES="$INCLUDES $ALSA_CFLAGS"; LIBS="$LIBS $ALSA_LIBS"],
+    [INCLUDES="${INCLUDES} $ALSA_CFLAGS"; LIBS="${LIBS} $ALSA_LIBS"],
     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([ENCA],  [enca],
-    [INCLUDES="$INCLUDES $ENCA_CFLAGS"; LIBS="$LIBS $ENCA_LIBS"],
+    [INCLUDES="${INCLUDES} $ENCA_CFLAGS"; LIBS="${LIBS} $ENCA_LIBS"],
     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([XT],    [xt],
-    [INCLUDES="$INCLUDES $XT_CFLAGS"; LIBS="$LIBS $XT_LIBS"],
+    [INCLUDES="${INCLUDES} $XT_CFLAGS"; LIBS="${LIBS} $XT_LIBS"],
     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([XTST],  [xtst],
-    [INCLUDES="$INCLUDES $XTST_CFLAGS"; LIBS="$LIBS $XTST_LIBS"],
+    [INCLUDES="${INCLUDES} $XTST_CFLAGS"; LIBS="${LIBS} $XTST_LIBS"],
     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([XEXT],  [xext],
-    [INCLUDES="$INCLUDES $XEXT_CFLAGS"; LIBS="$LIBS $XEXT_LIBS"],
+    [INCLUDES="${INCLUDES} $XEXT_CFLAGS"; LIBS="${LIBS} $XEXT_LIBS"],
     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([XMU],   [xmu],
-    [INCLUDES="$INCLUDES $XMU_CFLAGS"; LIBS="$LIBS $XMU_LIBS"],
+    [INCLUDES="${INCLUDES} $XMU_CFLAGS"; LIBS="${LIBS} $XMU_LIBS"],
     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([XINERAMA],[xinerama],
-    [INCLUDES="$INCLUDES $XINERAMA_CFLAGS"; LIBS="$LIBS $XINERAMA_LIBS"],
+    [INCLUDES="${INCLUDES} $XINERAMA_CFLAGS"; LIBS="${LIBS} $XINERAMA_LIBS"],
     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([CURL],  [libcurl],
-    [INCLUDES="$INCLUDES $CURL_CFLAGS"; LIBS="$LIBS $CURL_LIBS"],
+    [INCLUDES="${INCLUDES} $CURL_CFLAGS"; LIBS="${LIBS} $CURL_LIBS"],
     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([DBUS],  [dbus-1],
-    [INCLUDES="$INCLUDES $DBUS_CFLAGS"; LIBS="$LIBS $DBUS_LIBS"],
+    [INCLUDES="${INCLUDES} $DBUS_CFLAGS"; LIBS="${LIBS} $DBUS_LIBS"],
     AC_MSG_ERROR($missing_library))
   PKG_CHECK_MODULES([SDL],   [sdl],
-    [INCLUDES="$INCLUDES $SDL_CFLAGS"; LIBS="$LIBS $SDL_LIBS"],
+    [INCLUDES="${INCLUDES} $SDL_CFLAGS"; LIBS="${LIBS} $SDL_LIBS"],
     AC_MSG_ERROR($missing_library))
 fi
 
@@ -754,10 +754,10 @@
 else
   if test "$use_hal" = "yes"; then
     PKG_CHECK_MODULES([HAL], [hal],
-      [INCLUDES="$INCLUDES $HAL_CFLAGS"; LIBS="$LIBS $HAL_LIBS"],
+      [INCLUDES="${INCLUDES} $HAL_CFLAGS"; LIBS="${LIBS} $HAL_LIBS"],
       use_hal=no;AC_MSG_RESULT($hal_not_found))
     PKG_CHECK_MODULES([HAL_STORAGE], [hal-storage],
-      [INCLUDES="$INCLUDES $HAL_STORAGE_CFLAGS"; LIBS="$LIBS $HAL_STORAGE_LIBS"],
+      [INCLUDES="${INCLUDES} $HAL_STORAGE_CFLAGS"; LIBS="${LIBS} $HAL_STORAGE_LIBS"],
       use_hal=no;AC_MSG_RESULT($halstorage_not_found))
   else
     AC_MSG_RESULT($hal_disabled)
@@ -836,7 +836,7 @@
 # External FFmpeg
 if test "$use_external_ffmpeg" = "yes"; then
   PKG_CHECK_MODULES([FFMPEG], [libavcodec libavformat libavutil libpostproc libswscale],
-                    [INCLUDES="$INCLUDES $FFMPEG_CFLAGS"; LIBS="$LIBS $FFMPEG_LIBS"],
+                    [INCLUDES="${INCLUDES} $FFMPEG_CFLAGS"; LIBS="${LIBS} $FFMPEG_LIBS"],
                     AC_MSG_ERROR($missing_library))
 
   # Determine whether AVPacket and relevant functions are defined in libavformat
@@ -1055,7 +1055,7 @@
       AC_DEFINE([HAVE_LIBOPENMAX], [1], [Define to 1 if you have the 'OpenMax' library.]),
       use_openmax=no;AC_MSG_RESULT($openmax_not_found))
     if test "$use_openmax" = "yes"; then
-      INCLUDES="$INCLUDES -I/usr/include/OpenMAX/IL"
+      INCLUDES="${INCLUDES} -I/usr/include/OpenMAX/IL"
     fi
   else
     AC_MSG_NOTICE($openmax_disabled)
@@ -1219,9 +1219,9 @@
 fi
 
 if test -d "/usr/include/lzo"; then
-  INCLUDES="$INCLUDES -I/usr/include/lzo"
+  INCLUDES="${INCLUDES} -I/usr/include/lzo"
 else
-  INCLUDES="$INCLUDES"
+  INCLUDES="${INCLUDES}"
 fi
 
 if test "x$use_ccache" != "xno"; then
